test: set `ReadWritePaths=` only for units from TEST-20

mrc0mmand opened the pull request:
as setting it system-wide has some undesirable side-effects.

Follow-up to b6fc524.

--- 

I tested this manually with _all_ test cases w/ coverage enabled and it seems to finally work with all of them. So this is definitely the last coverage change for now, because *it's getting dark and my sanity is low*.

/cc @bluca

bluca commented:
Doesn't look good:

```
TEST-01-BASIC:                      SUCCESS  ( 89 s)
TEST-02-UNITTESTS:                  SUCCESS  (202 s)
TEST-03-JOBS:                       SUCCESS  ( 15 s)
TEST-04-JOURNAL:                    SUCCESS  ( 20 s)
TEST-05-RLIMITS:                    SUCCESS  (  3 s)
TEST-06-SELINUX:                    SUCCESS  (  0 s)
TEST-07-ISSUE-1981:                 SUCCESS  (  4 s)
TEST-08-ISSUE-2730:                 SUCCESS  (350 s)
TEST-09-ISSUE-2691:                 SUCCESS  (214 s)
TEST-10-ISSUE-2467:                 SUCCESS  ( 19 s)
TEST-11-ISSUE-3166:                 SUCCESS  (226 s)
TEST-12-ISSUE-3171:                 SUCCESS  ( 11 s)
TEST-13-NSPAWN-SMOKE:               FAIL     (1226 s)
TEST-14-MACHINE-ID:                 SUCCESS  (212 s)
TEST-15-DROPIN:                     SUCCESS  ( 69 s)
TEST-16-EXTEND-TIMEOUT:             SUCCESS  ( 67 s)
TEST-17-UDEV:                       FAIL     (804 s)
TEST-18-FAILUREACTION:              SUCCESS  (  8 s)
TEST-19-DELEGATE:                   SUCCESS  (173 s)
TEST-20-MAINPIDGAMES:               SUCCESS  (  9 s)
TEST-22-TMPFILES:                   SUCCESS  (  9 s)
TEST-23-TYPE-EXEC:                  SUCCESS  (  4 s)
TEST-24-CRYPTSETUP:                 SUCCESS  (398 s)
TEST-26-SETENV:                     SUCCESS  (  5 s)
TEST-27-STDOUTFILE:                 SUCCESS  (  4 s)
TEST-28-PERCENTJ-WANTEDBY:          SUCCESS  ( 10 s)
TEST-29-PORTABLE:                   FAIL     (344 s)
TEST-30-ONCLOCKCHANGE:              FAIL     (291 s)
TEST-31-DEVICE-ENUMERATION:         SUCCESS  (200 s)
TEST-32-OOMPOLICY:                  SUCCESS  (212 s)
TEST-33-CLEAN-UNIT:                 SUCCESS  (  9 s)
TEST-34-DYNAMICUSERMIGRATE:         SUCCESS  ( 23 s)
TEST-36-NUMAPOLICY:                 FAIL     (1220 s)
TEST-37-RUNTIMEDIRECTORYPRESERVE:   SUCCESS  ( 13 s)
TEST-38-FREEZER:                    FAIL     (598 s)
TEST-39-EXECRELOAD:                 SUCCESS  (  7 s)
TEST-40-EXEC-COMMAND-EX:            SUCCESS  (  6 s)
TEST-41-ONESHOT-RESTART:            SUCCESS  ( 14 s)
TEST-42-EXECSTOPPOST:               SUCCESS  ( 30 s)
TEST-43-PRIVATEUSER-UNPRIV:         SUCCESS  (  4 s)
TEST-44-LOG-NAMESPACE:              SUCCESS  (  7 s)
TEST-46-HOMED:                      SUCCESS  (  8 s)
TEST-47-ISSUE-14566:                SUCCESS  ( 14 s)
TEST-48-START-STOP-NO-RELOAD:       SUCCESS  ( 21 s)
TEST-49-RUNTIME-BIND-PATHS:         SUCCESS  ( 15 s)
TEST-50-DISSECT:                    FAIL     (629 s)
TEST-51-ISSUE-16115:                SUCCESS  ( 10 s)
TEST-52-HONORFIRSTSHUTDOWN:         SUCCESS  (  5 s)
TEST-53-ISSUE-16347:                SUCCESS  (221 s)
TEST-54-CREDS:                      SUCCESS  (  9 s)
TEST-55-OOMD:                       SUCCESS  ( 10 s)
TEST-56-EXIT-TYPE:                  SUCCESS  (  1 s)
TEST-57-ONSUCCESS-UPHOLD:           SUCCESS  ( 17 s)
TEST-58-REPART:                     FAIL     (305 s)
TEST-59-RELOADING-RESTART:          SUCCESS  ( 17 s)
TEST-60-MOUNT-RATELIMIT:            SUCCESS  (130 s)
TEST-62-RESTRICT-IFACES:            SUCCESS  (202 s)
TEST-63-ISSUE-17433:                SUCCESS  ( 11 s)
TEST-64-UDEV-STORAGE:               SUCCESS  (  1 s)
TEST-65-ANALYZE:                    SUCCESS  ( 10 s)
TEST-66-DEVICE-ISOLATION:           SUCCESS  (473 s)
TEST-67-INTEGRITY:                  FAIL     (620 s)
TEST-68-PROPAGATE-EXIT-STATUS:      SUCCESS  ( 11 s)
TEST-69-SHUTDOWN:                   SUCCESS  ( 26 s)
```

mrc0mmand commented:
On ppc64le only TEST-17 failed, which is expected.

As for the amd64 job, it looks like the hypervisor was overloaded?

TEST-29-PORTABLED
```
[  191.967110] systemd[1]: dev-ttyS0.device: Job dev-ttyS0.device/start timed out.
[  191.982297] systemd[1]: dev-ttyS0.device: Job 71 dev-ttyS0.device/start finished, result=timeout
[  191.983420] systemd[1]: Timed out waiting for device /dev/ttyS0.
[  192.000878] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=JobRemoved cookie=356 reply_cookie=0 signature=uoss error-name=n/a error-messa>
[  192.002151] systemd[1]: serial-getty@ttyS0.service: Job 69 serial-getty@ttyS0.service/start finished, result=dependency
[  192.002904] systemd[1]: Dependency failed for Serial Getty on ttyS0.
[  192.003819] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=JobRemoved cookie=357 reply_cookie=0 signature=uoss error-name=n/a error-messa>
[  192.010092] systemd[1]: serial-getty@ttyS0.service: Job serial-getty@ttyS0.service/start failed with result 'dependency'.
[  192.011332] systemd[1]: dev-ttyS0.device: Job dev-ttyS0.device/start failed with result 'timeout'.
...
[  293.425785] systemd[1]: minimal-app0-foo.service: start-pre operation timed out. Terminating.
[  293.456823] systemd[1]: minimal-app0-foo.service: Changed start-pre -> stop-sigterm
[  293.458458] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/minimal_2dapp0_2dfoo_2eservice interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=922 reply_cookie=0>
[  293.459440] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/minimal_2dapp0_2dfoo_2eservice interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=923 reply_cookie=0>
[  293.497053] systemd[319]: Remounted /run/systemd/unit-root/etc/resolv.conf.
[  293.498639] systemd[1]: Received SIGCHLD from PID 318 ((cat)).
[  293.499695] systemd[1]: Child 318 ((cat)) died (code=killed, status=15/TERM)
[  293.517918] systemd[1]: minimal-app0-foo.service: Child 318 belongs to minimal-app0-foo.service.
[  293.519281] systemd[1]: minimal-app0-foo.service: Control process exited, code=killed, status=15/TERM
[  293.554550] systemd[1]: minimal-app0-foo.service: Got final SIGCHLD for state stop-sigterm.
[  293.556659] systemd[1]: minimal-app0-foo.service: Failed with result 'timeout'.
[  293.601692] systemd[1]: minimal-app0-foo.service: Service will not restart (restart setting)
[  293.602958] systemd[1]: minimal-app0-foo.service: Changed stop-sigterm -> failed
[  293.682384] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=924 reply_cookie=0 signature=sa{sv}as error-name=n/a e>
[  293.683765] systemd[1]: minimal-app0-foo.service: Job 180 minimal-app0-foo.service/start finished, result=failed
[  293.690077] systemd[1]: Failed to start minimal-app0-foo.service.
...
[  294.341375] portablectl[294]: Job for minimal-app0-foo.service failed because a timeout was exceeded.
[  294.342712] sh[349]: + systemctl poweroff --no-block
[  294.441473] portablectl[294]: See "systemctl status minimal-app0-foo.service" and "journalctl -xeu minimal-app0-foo.service" for details.
[  294.442901] portablectl[294]: Unexpected job result, assuming server side newer than us: failed
[  294.540986] systemd[1]: Sent message type=method_return sender=n/a destination=:1.2 path=n/a interface=n/a member=n/a cookie=931 reply_cookie=7 signature=v error-name=n/a error-message=n/a
[  294.671067] portablectl[294]: Got result failed/Input/output error for job minimal-app0-foo.service
[  294.671956] systemd[1]: minimal-app0.service: start-pre operation timed out. Terminating.
[  294.673702] systemd[1]: minimal-app0.service: Changed start-pre -> stop-sigterm
[  294.674897] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/minimal_2dapp0_2eservice interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=932 reply_cookie=0 signa>
[  294.777451] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1/unit/minimal_2dapp0_2eservice interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=933 reply_cookie=0 signa>
[  294.779198] systemd[1]: systemd-journald.service: Received EPOLLHUP on stored fd 46 (stored), closing.
[  294.780631] systemd[1]: Received SIGCHLD from PID 319 ((cat)).
[  294.781443] systemd[1]: Child 319 ((cat)) died (code=killed, status=15/TERM)
[  294.783208] systemd[1]: minimal-app0.service: Child 319 belongs to minimal-app0.service.
[  294.874273] systemd[1]: minimal-app0.service: Control process exited, code=killed, status=15/TERM
[  294.875676] systemd[1]: minimal-app0.service: Got final SIGCHLD for state stop-sigterm.
[  294.913512] systemd[1]: minimal-app0.service: Failed with result 'timeout'.
[  294.915493] systemd[1]: minimal-app0.service: Service will not restart (restart setting)
[  294.916931] systemd[1]: minimal-app0.service: Changed stop-sigterm -> failed
[  294.917790] systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=934 reply_cookie=0 signature=sa{sv}as error-name=n/a e>
[  294.918689] systemd[1]: minimal-app0.service: Job 233 minimal-app0.service/start finished, result=failed
[  294.919479] systemd[1]: Failed to start minimal-app0.service.
```

TEST-30-ONCLOCKCHANGE:
```
[  194.179397] systemd-logind[268]: Watching system buttons on /dev/input/event0 (Power Button)
[  194.551701] testsuite-30.sh[271]: + timedatectl set-timezone Europe/Berlin
[  197.370678] dbus-daemon[290]: [system] AppArmor D-Bus mediation is enabled
[  198.206095] dbus-daemon[290]: [system] Successfully activated service 'org.freedesktop.systemd1'
[  198.345481] dbus-daemon[290]: [system] Activating via systemd: service name='org.freedesktop.timedate1' unit='dbus-org.freedesktop.timedate1.service' requested by ':1.1' (uid=0 pid=291 comm="timedatectl set-timezone Europe/Berlin " lab>
[  198.897617] audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-logind comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[  198.852651] kernel: audit: type=1130 audit(1645048757.023:18): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-logind comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[  206.025864] systemd-logind[268]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 keyboard)
[  223.204763] timedatectl[291]: Failed to set time zone: Connection timed out
[  223.994290] dbus-daemon[290]: [system] Failed to activate service 'org.freedesktop.timedate1': timed out (service_start_timeout=25000ms)
```

TEST-36-NUMAPOLICY:
```
[  245.431285] systemd[1]: dev-ttyS0.device: Job dev-ttyS0.device/start timed out.
[  245.559895] systemd[1]: dev-ttyS0.device: Job 69 dev-ttyS0.device/start finished, result=timeout
[  245.561669] systemd[1]: Timed out waiting for device /dev/ttyS0.
[  245.582270] testsuite-36.sh[272]: + stopStrace
[  245.630139] testsuite-36.sh[272]: + [[ -v COPROC_PID ]]
[  245.674286] testsuite-36.sh[272]: + local PID=278
[  245.716519] systemd[1]: serial-getty@ttyS0.service: Job 67 serial-getty@ttyS0.service/start finished, result=dependency
[  245.723994] systemd[1]: Dependency failed for Serial Getty on ttyS0.
[  245.754533] testsuite-36.sh[272]: + kill -s TERM 278
[  245.795229] testsuite-36.sh[272]: + kill -0 278
[  245.834317] testsuite-36.sh[272]: + sleep 0.1
[  245.862458] systemd[1]: serial-getty@ttyS0.service: Job serial-getty@ttyS0.service/start failed with result 'dependency'.
[  246.025244] systemd[1]: dev-ttyS0.device: Job dev-ttyS0.device/start failed with result 'timeout'.
[  246.053448] systemd[1]: systemd-logind.service: start operation timed out. Terminating.
[  246.054903] systemd[1]: systemd-logind.service: Changed start -> stop-sigterm
[  246.055756] systemd[1]: systemd-logind.service: Got notification message from PID 269 (STOPPING=1, STATUS=Shutting down...)
[  246.069621] systemd[1]: Bus private-bus-connection: changing state CLOSING → CLOSED
[  246.070737] systemd[1]: Got disconnect on private connection.
[  246.130625] systemd[1]: Received SIGCHLD from PID 269 (systemd-logind).
[  246.134431] systemd[1]: Child 269 (systemd-logind) died (code=exited, status=0/SUCCESS)
[  246.138259] systemd[1]: systemd-logind.service: Child 269 belongs to systemd-logind.service.
[  246.148563] systemd[1]: systemd-logind.service: Main process exited, code=exited, status=0/SUCCESS (success)
[  246.150558] systemd[1]: systemd-logind.service: Failed with result 'timeout'.
[  246.162516] systemd[1]: systemd-logind.service: Service will restart (restart setting)
[  246.169648] systemd[1]: systemd-logind.service: Changed stop-sigterm -> failed
[  246.186109] systemd[1]: systemd-logind.service: Job 72 systemd-logind.service/start finished, result=failed
[  246.211191] systemd[1]: Failed to start User Login Management.
...
E: Test timed out after 1200s
```

similarly for the other tests.

mrc0mmand commented:
@bluca I'd say all the fails are unrelated, but just to be on the safe side - could you re-trigger the amd64 job please?

bluca commented:
done - yeah it looks unrelated, but I've never seen _all_ of them fail at the same time, better to have another run just to be safe

mrc0mmand commented:
Yup, the amd64 job is green, so it indeed was a fluke.
